sanity_html/renderer.py

```diff
@@ -73,19 +76,17 @@ def _render_node(self, node: dict, context: Optional[Block] = None, list_item: b
         :param list_item: Whether we are handling a list upstream (impacts block handling).
         """
         if is_list(node):
+            logger.debug('Rendering node as list')
             block = Block(**node, marker_definitions=self._custom_marker_definitions)
             return self._render_list(block, context)
         elif is_block(node):
+            logger.debug('Rendering node as block')
             block = Block(**node, marker_definitions=self._custom_marker_definitions)
             return self._render_block(block, list_item=list_item)
 
         elif is_span(node):
-            if isinstance(node, str):
-                # TODO: Remove if we there's no coverage for this after we've fixed tests
-                #  not convinced this code path is possible - put it in because the sanity lib checks for it
-                span = Span(**{'text': node})
-            else:
-                span = Span(**node)
+            logger.debug('Rendering node as span')
+            span = Span(**node)
```

Comment:
Now that we've resolved all but one remaining test, I think it's time to get rid of this extra check. Doesn't seem useful.